theano/tensor/opt.py

```diff
@@ -2591,6 +2591,23 @@ def local_adv_sub1_adv_inc_sub1(node):
     return [T.cast(y, node.outputs[0].dtype)]
 
 
+@register_specialize
+@register_stabilize
+@register_canonicalize
+@gof.local_optimizer([IncSubtensor, AdvancedIncSubtensor1])
```

Comment:
If it can't work for that op, add a note to tell why so that we don't spend time looking at this case again.